### ISSUE TITLE
Server started to reboot constantly on savegame

### ISSUE THREAD

**Issue by Tuumke** [wolveix/satisfactory-server#238]
**Describe the Bug**
As of this morning, my savegame seems 'corrupt'? Server just keeps crashing. So i've been trying to figure out what it is, without any luck.
I decided to go and make a "new server" (backup current files, remove everything, restart container), but that just disconnects me and restarts the docker as well

**Your Runtime Command or Docker Compose File**
```
networks:
  satisfactory:
    name: satisfactory
    driver: bridge
    ipam:
      config:
        - subnet: 192.168.100.0/24
services:
    satisfactory-server:
        container_name: 'satisfactory-server'
        hostname: 'satisfactory-server'
        image: 'wolveix/satisfactory-server:latest'
        networks:
          satisfactory:
            ipv4_address: 192.168.100.230
        ports:
            - '7777:7777/udp'
            - '15000:15000/udp'
            - '15777:15777/udp'
        volumes:
            - "$DOCKERDIR/games/satisfactory/config:/config"
        environment:
            - PGID=$PGID                  #set the group ID of the user the server will run as
            - PUID=$PUID                  #set the user ID of the user the server will run as
            - AUTOPAUSE=false             #pause game when no player is connected
            - AUTOSAVEINTERVAL=600        #autosave interval in seconds
            - AUTOSAVENUM=5               #number of rotating autosave files
            - AUTOSAVEONDISCONNECT=true   #autosave when last player disconnects
            - CRASHREPORT=true            #automatic crash reporting
            - DEBUG=false                 #for debugging the server
            - DISABLESEASONALEVENTS=false #disable the FICSMAS event (you miserable bastard)
            - MAXOBJECTS=2162688          #set the object limit for your server
            - MAXPLAYERS=6                #set the player limit for your server
            - MAXTICKRATE=30              #set the maximum sim tick rate for your server
            - NETWORKQUALITY=3            #set the network quality/bandwidth for your server
            - SERVERBEACONPORT=15000      #set the game's beacon port
            - SERVERGAMEPORT=7777         #set the game's port
            #- SERVERIP=0.0.0.0            #set the game's ip (usually not needed)
            - SERVERQUERYPORT=15777       #set the game's query port
            - SKIPUPDAT=false             #avoid updating the game on container start/restart
            - STEAMBETA=false             #set experimental game version
            - TIMEOUT=300                 #set client timeout (in seconds)
            - SERVERSTREAMING=false
        restart: unless-stopped
        deploy:
          resources:
            limits:
              memory: 24G
            reservations:
              memory: 12G
```

**System Specs (please complete the following information):**
If you're on Linux, just paste the following block as a single command, and paste the output here.
```
===== START ISSUE REPORT =====
OS:  Linux thebox 6.1.74-Unraid #1 SMP PREEMPT_DYNAMIC Fri Feb  2 11:06:32 PST 2024 x86_64 Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz GenuineIntel GNU/Linux
CPU: Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
BIOS Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz  CPU @ 3.5GHz
RAM: 55GB/64GB
HDD: 165GB/32GB (1% used)
===== END ISSUE REPORT =====

HDD is incorrect! I have 32TB in total ;-) 
```

**Additional Context**
Had been messing arround with the ini files to get some better network quality, see if we could get rid of the lag in the trains. I did this, after we reache milestone 8? Working our way to nuclear power. And now, the server just keeps restarting (client just disconnects after few seconds), docker logs -f satisfactory-server stops working and needs to execute that command again. So the container is actually restarting. I can't figure out why this is...

**Comment by Tuumke** [comment]
> Hey @Tuumke! Sounds fun :o Sorry to hear that you're experiencing this. How were you modifying the `.ini` files? Using the `/config/overrides`? Or relying on environment variables?
> 
> Could you please provide the full log from `docker logs satisfactory-server`? Thanks!
> 
> Also, FYI, your `SKIPUPDATE` environment variable is missing an E on the end :) It's false by default so it doesn't really matter, but it just caught my eye.

Still trying a few things out. I'm running it on my UnRAID machine, so it was using multiple disks. Now using my cache disk, but it looks actually slower then ever on that disk (which wasnt being used previously). Managed to load up a default desert start, now trying to load my savegame in there. I'll try to revert back to the mount it was using as place in my docker-compose and try to hand over the logs if it's not working. 

I managed to get the savegame running on a Nitrado host with all the same settings/ini files.

I was editing the INI files using [filebrowser ](https://github.com/filebrowser/filebrowser/) docker container. Can easily upload files as well with that (savegames after you've put them through SF Calculator for example).  

Will get back to you a.s.a.p.

**Comment by wolveix** [comment]
Gotcha. As an FYI, modifying the in-place `.ini` files won't have an effect, as they get regenerated on container start (unless you're modifying the source files, and only modifying values that we don't touch). You should use the `overrides` directory instead if you need to modify values that we don't have environment variables for

**Comment by Tuumke** [comment]
@wolveix i attached the file where i loaded up my savegame. It crashed at somepoint during loading. I also attached the log i got after it restarted where it seems to be stuck as well.

[after-automatic-restart-satisfactory-server_logs.txt](https://github.com/wolveix/satisfactory-server/files/14838083/after-automatic-restart-satisfactory-server_logs.txt)
[satisfactory-server_logs.txt](https://github.com/wolveix/satisfactory-server/files/14838085/satisfactory-server_logs.txt)

**Comment by Tuumke** [comment]
I also notice a lot of network changes in docker whenever this happens..

```
Apr  2 08:31:36 thebox kernel: br-676a178cd9fc: port 1(veth0f7177a) entered disabled state
Apr  2 08:31:36 thebox kernel: veth40cd0af: renamed from eth0
Apr  2 08:31:37 thebox kernel: br-676a178cd9fc: port 1(veth0f7177a) entered disabled state
Apr  2 08:31:37 thebox kernel: device veth0f7177a left promiscuous mode
Apr  2 08:31:37 thebox kernel: br-676a178cd9fc: port 1(veth0f7177a) entered disabled state
Apr  2 08:31:38 thebox kernel: veth16de645: renamed from eth0
Apr  2 08:31:38 thebox kernel: br-f8a7f7f71814: port 24(vethc9f3632) entered disabled state
Apr  2 08:31:39 thebox kernel: br-f8a7f7f71814: port 24(vethc9f3632) entered disabled state
Apr  2 08:31:39 thebox kernel: device vethc9f3632 left promiscuous mode
Apr  2 08:31:39 thebox kernel: br-f8a7f7f71814: port 24(vethc9f3632) entered disabled state
Apr  2 08:31:39 thebox kernel: veth43befba: renamed from eth1
Apr  2 08:31:39 thebox kernel: br-676a178cd9fc: port 2(veth15053a9) entered disabled state
Apr  2 08:31:39 thebox kernel: br-676a178cd9fc: port 2(veth15053a9) entered disabled state
Apr  2 08:31:39 thebox kernel: device veth15053a9 left promiscuous mode
Apr  2 08:31:39 thebox kernel: br-676a178cd9fc: port 2(veth15053a9) entered disabled state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered blocking state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered disabled state
Apr  2 08:31:40 thebox kernel: device veth174900a entered promiscuous mode
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered blocking state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered forwarding state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered disabled state
Apr  2 08:31:40 thebox kernel: eth0: renamed from vethc4e1153
Apr  2 08:31:40 thebox kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth174900a: link becomes ready
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered blocking state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 1(veth174900a) entered forwarding state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 2(veth8ae0bed) entered blocking state
Apr  2 08:31:40 thebox kernel: br-676a178cd9fc: port 2(veth8ae0bed) entered disabled state
Apr  2 08:31:40 thebox kernel: device veth8ae0bed entered promiscuous mode
Apr  2 08:31:41 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered blocking state
Apr  2 08:31:41 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered disabled state
Apr  2 08:31:41 thebox kernel: device vethb98fc5d entered promiscuous mode
Apr  2 08:31:41 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered blocking state
Apr  2 08:31:41 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered forwarding state
Apr  2 08:31:42 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered disabled state
Apr  2 08:31:42 thebox kernel: eth0: renamed from veth13c3f89
Apr  2 08:31:42 thebox kernel: IPv6: ADDRCONF(NETDEV_CHANGE): vethb98fc5d: link becomes ready
Apr  2 08:31:42 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered blocking state
Apr  2 08:31:42 thebox kernel: br-f8a7f7f71814: port 24(vethb98fc5d) entered forwarding state
Apr  2 08:31:42 thebox kernel: eth1: renamed from vethf32ddd0
Apr  2 08:31:42 thebox kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth8ae0bed: link becomes ready
Apr  2 08:31:42 thebox kernel: br-676a178cd9fc: port 2(veth8ae0bed) entered blocking state
Apr  2 08:31:42 thebox kernel: br-676a178cd9fc: port 2(veth8ae0bed) entered forwarding state
Apr  2 08:31:49 thebox kernel: veth3feb829: renamed from eth0
Apr  2 08:31:49 thebox kernel: br-f8a7f7f71814: port 22(vetha9ac95e) entered disabled state
Apr  2 08:31:49 thebox kernel: br-f8a7f7f71814: port 22(vetha9ac95e) entered disabled state
Apr  2 08:31:49 thebox kernel: device vetha9ac95e left promiscuous mode
Apr  2 08:31:49 thebox kernel: br-f8a7f7f71814: port 22(vetha9ac95e) entered disabled state
Apr  2 08:31:50 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered blocking state
Apr  2 08:31:50 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered disabled state
Apr  2 08:31:50 thebox kernel: device veth025bc47 entered promiscuous mode
Apr  2 08:31:50 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered blocking state
Apr  2 08:31:50 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered forwarding state
Apr  2 08:31:50 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered disabled state
Apr  2 08:31:51 thebox kernel: eth0: renamed from veth3c0a47b
Apr  2 08:31:51 thebox kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth025bc47: link becomes ready
Apr  2 08:31:51 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered blocking state
Apr  2 08:31:51 thebox kernel: br-f8a7f7f71814: port 22(veth025bc47) entered forwarding state
Apr  2 08:32:04 thebox kernel: br-18371025d7b1: port 1(vethebdaf23) entered disabled state
Apr  2 08:32:04 thebox kernel: veth0eb13ea: renamed from eth0
Apr  2 08:32:04 thebox kernel: br-18371025d7b1: port 1(vethebdaf23) entered disabled state
Apr  2 08:32:04 thebox kernel: device vethebdaf23 left promiscuous mode
Apr  2 08:32:04 thebox kernel: br-18371025d7b1: port 1(vethebdaf23) entered disabled state
Apr  2 08:32:05 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered blocking state
Apr  2 08:32:05 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered disabled state
Apr  2 08:32:05 thebox kernel: device vethc625a8d entered promiscuous mode
Apr  2 08:32:05 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered blocking state
Apr  2 08:32:05 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered forwarding state
Apr  2 08:32:05 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered disabled state
Apr  2 08:32:06 thebox kernel: eth0: renamed from veth5391f2c
Apr  2 08:32:06 thebox kernel: IPv6: ADDRCONF(NETDEV_CHANGE): vethc625a8d: link becomes ready
Apr  2 08:32:06 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered blocking state
Apr  2 08:32:06 thebox kernel: br-18371025d7b1: port 1(vethc625a8d) entered forwarding state

```

**Comment by wolveix** [comment]
This could be related to CSS's recent dedicated server changes. Try uncommenting your `SERVERIP` environment variable. Networking should not be changing in the way that you're seeing.

**Comment by Tuumke** [comment]
I think it's related to portainer. On the portainer stack i opted in for GitOps
```
Meet ‘GitOps updates’ : Formerly known as Automatic updates

We've renamed "Automatic updates" to better align with industry terminology and clarify its purpose for all users. Originally chosen during the early emergence of GitOps, the name has changed, but the functionality remains unchanged. GitOps has rapidly emerged as a revolutionary approach to managing infrastructure and application changes, and we want to ensure our platform reflects the latest advancements in the industry.
```

GitOps has 2 options being:
Re-pull image
Force redeployment

Both were set to true/enabled. I disabled these and seems the servers has been running for the past hour or so.

I came to try this after i put 2 and 2 together, being:
- Server is not restarting, otherwise i wouldn't see those network changes in the syslog
- Docker container is being recreated, as the logs stop showing as well (docker completely disappears and is being recreated)

I have to check portainer to see if i can figure out why it keeps repulling as i'm not changing the compose or the entire ref for my main. (I'm pulling docker-compose from my private github repo, as i do with others too).
